### PR TITLE
bzl: enable race detection in tests

### PR DESCRIPTION
**Pull request by mikedanese**
blocked by the bazel build fix

### CODE REVIEW

**Comment by spxtr** [comment]
Could also consider putting it on [these lines](https://github.com/kubernetes/kubernetes/blob/6430b1002c2900a1761cbd07cb77a1e4cf82b64e/build/root/Makefile#L543) so that it's not just some magic environment. I'm happy either way though.

**Comment by ixdy** [comment]
do we want to enable race detection always? or only in CI?

File: build/root/.bazelrc [hunk 1]
```diff
@@ -11,3 +11,6 @@ build --sandbox_tmpfs_path=/tmp
 # Ensure that Bazel never runs as root, which can cause unit tests to fail.
 # This flag requires Bazel 0.5.0+
 build --sandbox_fake_username
+
+# Run all tests with race detector
+test --features=race
```
Comment (ixdy) [inline build/root/.bazelrc]: another possibility is making this
```
test:race --features=race
```
and then it'd only be enabled if you `bazel test --config=race`
(though I'm not sure how that'd interact with the Makefile)

Comment (spxtr) [inline build/root/.bazelrc]: At that point you may as well just say `--features=race` :P

**Comment by spxtr** [comment]
Nice, you hit a race already! https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/50792/pull-kubernetes-bazel/36234/

File: build/root/.bazelrc (hunk 1 above)
Comment (mikedanese) [inline build/root/.bazelrc]: interesting... https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/mutation_detector_test.go#L1

Comment (spxtr) [inline build/root/.bazelrc]: wat

**Comment by spxtr** [comment]
Does that test ever run in the non-bazel job currently? I thought that job has race detection enabled.

File: build/root/.bazelrc (hunk 1 above)
Comment (spxtr) [inline build/root/.bazelrc]: https://github.com/kubernetes/kubernetes/blob/master/pkg/master/master_openapi_test.go

**Pull request closed without merging**